extensions/git/src/git.ts

```diff
@@ -334,7 +334,7 @@ export class Git {
 	}
 
 	async clone(url: string, parentPath: string, cancellationToken?: CancellationToken): Promise<string> {
-		let baseFolderName = decodeURI(url).replace(/^.*\//, '').replace(/\.git$/, '') || 'repository';
+		let baseFolderName = path.win32.basename(decodeURI(url), '.git') || 'repository';
```

Comment:
This won't work for URLs with forward slashes (the most common use case).

How about instead of `.replace(/^.*\//, '')` we do `.replace(/^.*[\/\\]/, '')`?

Reply:
I'll run some test cases on this, thanks!

Reply:
@joaomoreno this may be a newbie issue on the project.  I've looked through issues in the past to try and resolve this as well to no avail.  When I run the 'Debug: Git Unit Tests' launch configuration I get a "Cannot find module 'vscode'".  I can't seem to run the mocha tests for the git extension.  I was wanting to add tests to this pr if possible.

Anyways, both your Regex and the path.win32.basename seem to work fine.  I tried these paths against both fixes.  The baseFolderName just needs to resolve the name of the directory to clone the repo into.
```
https://foo/bar.git <-- resolves as 'bar'
T:\__SharedAssets\Assets_Houdini_Dev\houdini16.5\python2.7libs <-- resolves as 'python2.7libs'
/home/bob.git <-- resolves as 'bob'
```
Let me know if there is a path I may not have addressed, I can commit the regex update with tests if I can resolve that issue above.

Reply:
The test suite might be broken now. Just push the change, we´ll take it.

Reply:
It is pushed, thanks João!